src/pages/blog/coming-soon-threading.md

```diff
@@ -0,0 +1,119 @@
+
+### Setting \[conductor\]sync\_power\_state\_workers
+
+The number of threads which are launched for synchronizing the power state is controlled by the *sync_power_state_workers* option. A constraint is that the *periodic_max_workers* setting must be at least the same value, otherwise the workers may be restricted. If you have slow responding Baseboard Management Controllers, you may want to increase these settings.
+
+### Setting \[conductor\]sync\_power\_state\_interval
+
+The interval, in seconds, between launching new attempts to check and synchronize the power state of remote hosts with the database. This option, by default, is set to every sixty seconds. While multiple sweeps cannot run at the same time, the default setting may be a bit aggressive for larger clusters.
+
+### Setting \[sensor\_data\]workers
+
+This is the maximum concurrency for sensor data collection by a conductor. Sensor data collection, depending on the management protocol or even the performance of the remote BMC, can wildly impact the overall performance of this process. For example, a single host may take 30-60 seconds to query overall. Obviously, if your collecting sensor data, you may want to increase the worker count, and appropriately set the frequency in which you collect sensor data.
+
+### Timeout and Retry Settings
+
+A trend we have seen over the years is some operators wish to avoid all possible failures by extending timeouts. This is extremely counter-intuitive because timeouts should generally be shorter, as each thread stuck waiting to timeout is a consumed thread. If your trying to tune, do take a look at your timeout settings and consider 30 seconds instead of 60 seconds. Consider less retries in general. 
+
+## The Future Beyond Threading
+
+As a project, we recognize we have two distinct challenges which threading helps, but also doesn't help. Blocking IO, and the thundering herd of periodic tasks.
+
+Moving forward, we expect to take a more hybrid approach where we are using coroutines in threads to increase the parallelism upon which we perform certain actions like power state sync or sensor data collection.
+
+For periodic tasks, we're likely to move to a model where we keep a list of delayed future actions which need to occur, and utilize a single periodic task to spawn these actions as follow-ups. That said, both of these changes are substantial changes beyond the removal of eventlet. In a weird sense, they are actually more complex changes to ironic's architecture than just removing eventlet and getting threading as a side effect.
+
+## Thanks
+
+A special thanks goes to all of the Ironic contributors who have worked some long hours over this past development cycle to help ensure we can remove eventlet. We're almost there, and hopefully it will be gone in the next few weeks!
```

Comment:
Any value in linking to https://removal.eventlet.org/ or mentioning the larger OpenStack effort? A lot of people who use Ironic also use pieces which will likely remain eventlet-d for a while.